Modules/_localemodule.c

```diff
@@ -87,6 +87,36 @@ copy_grouping(const char* s)
     return result;
 }
 
+#if defined(MS_WINDOWS)
+static int
+check_locale_name(const char *locale, const char *end)
+{
+    size_t len = end ? (size_t)(end - locale) : strlen(locale);
+    const char *dot = memchr(locale, '.', len);
+    if (dot && locale + len - dot > 16) {
+        return -1;
```

Comment:
Where did you get the "16" from ?
Please include some comments to explain where you got it from and why this check is necessary. Thanks.

Reply:
BTW: It's better to make this limit configurable via a constant.

Reply:
Comes from https://github.com/huangqinjin/ucrt/blob/d6e817a4cc90f6f1fe54f8a0aa4af4fff0bb647d/inc/corecrt_internal.h#L401, though I don't think that constant is available for us at compile time (the internal headers for the CRT are, well, internal).

A constant at least gives it a name though, and using `MAX_CP_LEN` may help someone find an updated definition in the future.

Reply:
Looks good now.